### PR TITLE
[WIP] Check test data

### PR DESCRIPTION
**Pull request by choltz95**
Paper trail for #130 Following sklearn convention, added _check_test_data function with support for enforcing equal dimension at test-time. Let me know if this looks okay!

### CODE REVIEW

File: tslearn/svm.py [hunk 1]
```diff
@@ -71,6 +71,15 @@ def _preprocess_sklearn(self, X, y=None, fit_time=False):
         else:
             return sklearn_X, y
 
+    def _check_test_data(self, X):
+        n_samples, n_seqlen, n_features = X.shape
```
Comment (johannfaouzi) [inline tslearn/svm.py]: You can take the last dimension only if you use only this one, as you did it in the next line.

File: tslearn/svm.py [hunk 2]
```diff
@@ -71,6 +71,15 @@ def _preprocess_sklearn(self, X, y=None, fit_time=False):
...
+        expected_n_features = self._X_fit.shape[2]
+        if not n_features == expected_n_features:
+            raise ValueError("Incorrect number of features. "
+                             "Got %d features, expected %d" % (
+                                 n_features, expected_n_features))
+
+        return X
```
Comment (johannfaouzi) [inline tslearn/svm.py]: I think that it should not return anything since `X` is never modified.

**Comment by rtavenar** [comment]
@choltz95 Could you implement @johannfaouzi 's suggested change (there is no hurry, just to know if you can do it or not)?

**Comment by choltz95** [comment]
Yes! - will finish it up this week/weekend.

**Comment by rtavenar** [comment]
I think I can close this one since this issue is now investigated in #223

**Pull request closed without merging**